src/sinks/influxdb_metrics.rs

```diff
@@ -0,0 +1,916 @@
+                ("quantile_0.95".to_owned(), Field::Float(val)),
+            ]
+            .into_iter()
+            .collect(),
+        );
+    }
+
+    samples.sort_by(|a, b| a.partial_cmp(b).unwrap_or(Ordering::Equal));
+
+    let length = samples.len() as f64;
+    let min = samples.first().unwrap();
+    let max = samples.last().unwrap();
+
+    let p50 = samples[(0.50 * length - 1.0).round() as usize];
+    let p95 = samples[(0.95 * length - 1.0).round() as usize];
+
+    let sum = samples.iter().sum();
+    let avg = sum / length;
+
+    let fields: HashMap<String, Field> = vec![
+        ("min".to_owned(), Field::Float(*min)),
+        ("max".to_owned(), Field::Float(*max)),
+        ("median".to_owned(), Field::Float(p50)),
+        ("avg".to_owned(), Field::Float(avg)),
+        ("sum".to_owned(), Field::Float(sum)),
+        ("count".to_owned(), Field::Float(length)),
+        ("quantile_0.95".to_owned(), Field::Float(p95)),
+    ]
+    .into_iter()
+    .collect();
+
+    Some(fields)
+}
+
+// https://v2.docs.influxdata.com/v2.0/reference/syntax/line-protocol/
+fn influx_line_protocol(
+    measurement: String,
+    metric_type: &str,
+    tags: Option<HashMap<String, String>>,
+    fields: Option<HashMap<String, Field>>,
+    timestamp: i64,
+) -> String {
+    let mut line_protocol = vec![encode_key(measurement)];
+
+    // Tags
+    let mut unwrapped_tags = tags.unwrap_or(HashMap::new());
+    unwrapped_tags.insert("metric_type".to_owned(), metric_type.to_owned());
+    line_protocol.push(format!(",{}", encode_tags(unwrapped_tags)));
+
+    // Fields
+    let unwrapped_fields = fields.unwrap_or(HashMap::new());
+    let encoded_fields = encode_fields(unwrapped_fields);
+    if encoded_fields.is_empty() {
+        return "".to_owned();
+    }
+    line_protocol.push(format!(" {}", encoded_fields));
+
+    // Timestamp
+    line_protocol.push(format!(" {}", timestamp));
+
+    line_protocol.join("")
+}
+
+fn encode_key(key: String) -> String {
+    key.replace("\\", "\\\\")
+        .replace(",", "\\,")
+        .replace(" ", "\\ ")
+        .replace("=", "\\=")
+}
+
+fn encode_tags(tags: HashMap<String, String>) -> String {
+    let ordered: Vec<String> = tags
+        // sort by key
+        .iter()
+        .collect::<BTreeMap<_, _>>()
+        // map to key=value
+        .iter()
+        .map(|pair| {
+            let key = encode_key(pair.0.to_string());
+            let value = encode_key(pair.1.to_string());
+            if !key.is_empty() && !value.is_empty() {
+                format!("{}={}", key, value)
+            } else {
+                "".to_string()
+            }
+        })
+        // filter empty
+        .filter(|tag_value| !tag_value.is_empty())
+        .collect();
+
+    ordered.join(",")
+}
+
+fn encode_fields(fields: HashMap<String, Field>) -> String {
+    let encoded = fields
+        // sort by key
+        .iter()
+        .collect::<BTreeMap<_, _>>()
```

Comment:
It's just for consistency with tags? Should I change it?

Reply:
If InfluxDB is fine with getting an unsorted list of fields, it would improve performance of Vector if this sorting was removed and instead the `for` loop was called directly on `fields.iter()`.

Reply:
Done